### PR TITLE
#6911: Share coordinates fix

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR fixes Share plugin always places marker in the center of the map when defaultEnabled prop is `markerAndZoom` and opened from floating toolbar which point marked by GFI

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**
#6911 

**What is the new behavior?**
- Share function will retain the position of marker of GFI when opened from toolbar (floating toolbar)
- Share function will behave as is for other scenarios
Config for share plugin
```
{
  "name": "Share",
  "cfg": {
    "advancedSettings": {
      "bbox": true,
      "centerAndZoom": true,
      "defaultEnabled": "markerAndZoom"
    }
  },
  "override": {
    "Toolbar": {
      "priority": 1
    },
    "BurgerMenu": {
      "priority": 1
    }
  }
}
```

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
Issue ref: https://github.com/geosolutions-it/austrocontrol-C125/issues/287

### CODE REVIEW

File: web/client/epics/queryparams.js [hunk 1]
```diff
@@ -164,16 +165,30 @@ export const disableGFIForShareEpic = (action$, { getState = () => { } }) =>
     action$.ofType(TOGGLE_CONTROL)
         .filter(({control}) => control === "share")
         .switchMap(() => {
-            const shareEnabled = get(getState(), 'controls.share.enabled');
-            const mapInfoEnabled = get(getState(), 'mapInfo.enabled');
+            const state = getState();
+            const shareEnabled = get(state, 'controls.share.enabled');
+            const mapInfoEnabled = get(state, 'mapInfo.enabled');
```
Comment (MV88) [inline web/client/epics/queryparams.js:170]: there should be a selector for this, see mapInfoDisabledSelector

**Review by MV88 — CHANGES_REQUESTED** [review]
hi, the pr overall is good, just a few comments here and there

File: web/client/plugins/Share.jsx [hunk 2]
```diff
@@ -23,8 +23,8 @@ import { mapSelector } from '../selectors/map';
 import { currentContextSelector } from '../selectors/context';
 import { get } from 'lodash';
 import controls from '../reducers/controls';
-import {featureInfoClick, changeFormat, hideMapinfoMarker} from '../actions/mapInfo';
-import { addMarker } from '../actions/search';
+import { changeFormat } from '../actions/mapInfo';
+import { addMarker, hideMarker } from '../actions/search';
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/client/plugins/Share.jsx:27]: Yes. That's right

File: web/client/utils/CoordinatesUtils.js [hunk 3]
```diff
@@ -1030,6 +1030,25 @@ export const getPolygonFromCircle = (center, radius, units = "degrees", steps =
     return turfCircle(center, radius, {steps, units});
 };
 
+/**
+ * Generates longitude and latitude value from the point
+ * @param {object} point with latlng data
+ * @return {array} corrected longitude and latitude
+ */
+export const getLonLatFromPoint = (point) =>{
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/client/utils/CoordinatesUtils.js:1038]: Sure. I will add some

**Review by MV88 — APPROVED** [review]

**Pull request merged**